Sources/Commands/SwiftTestCommand.swift

```diff
@@ -774,13 +786,25 @@ extension SwiftTestCommand {
                     )
 
                     // Finally, run the tests.
+                    var output = [String]()
                     let result = runner.test(outputHandler: {
-                        // command's result output goes on stdout
-                        // ie "swift test" should output to stdout
-                        print($0, terminator: "")
+                        output.append($0)
                     })
                     if result == .failure {
                         swiftCommandState.executionStatus = .failure
+                        // If the runner reports failure do a check to ensure
+                        // all the binaries are present on the file system.
+                        try testProducts.map(\.binaryPath).forEach {
```

Comment:
If an error is thrown here, does this mean that everything collected in `output` is lost?

Also, coding style nit: we prefer `for` loops instead of `forEach`.

Reply:
Done